src/backend/distributed/commands/table.c

```diff
@@ -939,6 +939,32 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 				 */
 				constraint->skip_validation = true;
 			}
+			else if (constraint->contype == CONSTR_PRIMARY)
+			{
+				if (constraint->conname == NULL)
+				{
+					Relation rel = RelationIdGetRelation(leftRelationId);
+					constraint->conname = ChooseIndexName(RelationGetRelationName(rel),
+														  RelationGetNamespace(rel),
+														  NULL, NULL, true, true);
+					RelationClose(rel);
+
+					Assert(GetMemoryChunkContext(constraint->conname) ==
+						   GetMemoryChunkContext(newCmd));
+
+					((Constraint *) (newCmd->def))->conname = pstrdup(
+						constraint->conname);
+
+					/*
+					 * We have to change ALTER TABLE ... ADD PRIMARY ... command into
+					 * ALTER TABLE ... ADD CONSTRAINT <conname> PRIMARY KEY ...
+					 * in order to be able to use the name we created. Therefore we will send the
+					 * changed command to workers.
+					 */
+					deparseAT = true;
+					useInitialDDLCommandString = true;
```

Comment:
So, here we don't want to use the initial DDL command string, so we should set this to `false`.
To clarify, the initial DDL command string is 
`ALTER TABLE tbl ADD PRIMARY KEY`
However, to the shell tables we want to send
`ALTER TABLE tbl ADD CONSTRAINT tbl_pkey PRIMARY KEY`

Reply:
If I do not set useInitialDDLCommandString , the coordinator will not send any command for the main table (neither the original nor the new command), thus it becomes broken. I only went with minimal change to make the scenario work in this iteration. But, this part of the function needs to be rewritten to make sense.

Reply:
Yes, I understand that with the current logic the metadata sync command will be NULL, but we should extend that part, not set this boolean to true.

If we just set `useInitialDDLcommand` to true it will not be meaningful for this boolean value.

Reply:
> I only went with minimal change to make the scenario work in this iteration. But, this part of the function needs to be rewritten for sure to make sense.

Got it, so I guess this is part of the draft. Makes sense then 👍